react/features/video-quality/subscriber.js

```diff
@@ -193,43 +195,109 @@ function _updateReceiverVideoConstraints({ getState }) {
     const maxFrameHeight = Math.min(maxReceiverVideoQuality, preferredVideoQuality);
     const { remoteScreenShares } = state['features/video-layout'];
     const { visibleRemoteParticipants } = state['features/filmstrip'];
+    const { flags: { sourceNameSignaling } } = state['features/base/config'];
+    const { local: { id: localParticipantId } } = state['features/base/participants'];
+    const tracks = state['features/base/tracks'];
 
     const receiverConstraints = {
         constraints: {},
         defaultConstraints: { 'maxHeight': VIDEO_QUALITY_LEVELS.NONE },
         lastN,
-        onStageEndpoints: [],
-        selectedEndpoints: []
+        ...sourceNameSignaling ? { prioritizedSources: [] } : { onStageEndpoints: [] },
+        ...sourceNameSignaling ? { selectedSources: [] } : { selectedEndpoints: [] }
     };
 
-    // Tile view.
-    if (shouldDisplayTileView(state)) {
-        if (!visibleRemoteParticipants?.size) {
-            return;
+    if (sourceNameSignaling) {
+        let visibleRemoteTrackSourceNames;
+        let largeVideoSourceName;
+
+        if (visibleRemoteParticipants?.size) {
+            visibleRemoteTrackSourceNames = [ ...visibleRemoteParticipants ].reduce((acc, participantId) => {
+                const sourceName = getTrackSourceNameByMediaTypeAndParticipant(tracks, MEDIA_TYPE.VIDEO, participantId);
+
+                if (sourceName) {
+                    acc.push(sourceName);
+                }
+
+                return acc;
+            }, []);
         }
 
-        visibleRemoteParticipants.forEach(participantId => {
-            receiverConstraints.constraints[participantId] = { 'maxHeight': maxFrameHeight };
-        });
+        if (localParticipantId !== largeVideoParticipantId) {
```

Comment:
I was running into a bug where `getSourceName` was `undefined` when there is only one participant or if the user select themselves for large video.